lib/SILOptimizer/Analysis/ArraySemantic.cpp

```diff
@@ -537,6 +538,7 @@ bool swift::ArraySemanticsCall::doesNotChangeArray() const {
     case ArrayCallKind::kGetCount:
     case ArrayCallKind::kGetCapacity:
     case ArrayCallKind::kGetElement:
+    case ArrayCallKind::kAppendContentsOf:
```

Comment:
It's not correct to say that kAppendContentsOf does not change _the_ array. What you care about is the _newElements_ argument, which is not changed (and not the self-array).
I would remove this here and make a special case where doesNotChangeArray is called (see below).